test/releasers/java-yoshi.ts

```diff
@@ -580,4 +580,158 @@ describe('JavaYoshi', () => {
+    // the appropriate changes are proposed:
+    let expectedChanges = null;
+    sandbox.replace(
+      suggester,
+      'createPullRequest',
+      (_octokit, changes): Promise<number> => {
+        expectedChanges = [...(changes as Map<string, object>)]; // Convert map to key/value pairs.
+        return Promise.resolve(22);
+      }
+    );
+    const versionsContent = readFileSync(
+      resolve(fixturesPath, 'pre-ga-versions.txt'),
+      'utf8'
+    );
+    const readmeContent = readFileSync(
+      resolve(fixturesPath, 'README.md'),
+      'utf8'
+    );
+    const pomContents = readFileSync(resolve(fixturesPath, 'pom.xml'), 'utf8');
+    const googleUtilsContent = readFileSync(
+      resolve(fixturesPath, 'GoogleUtils.java'),
+      'utf8'
+    ).replace(/\r\n/g, '\n');
+    const graphql = JSON.parse(
+      readFileSync(resolve(fixturesPath, 'commits-promote-ga.json'), 'utf8')
+    );
+    const req = nock('https://api.github.com')
+      // This step looks for release PRs that are already open:
+      .get('/repos/googleapis/java-trace/pulls?state=open&per_page=100')
+      .reply(200, [])
+      .get(
+        '/repos/googleapis/java-trace/pulls?state=closed&per_page=100&sort=merged_at&direction=desc'
+      )
+      .reply(200, undefined)
+      .get(
+        '/repos/googleapis/java-trace/contents/versions.txt?ref=refs/heads/master'
+      )
+      .reply(200, {
+        content: Buffer.from(versionsContent, 'utf8').toString('base64'),
+        sha: 'abc123',
+      })
+      // fetch semver tags, this will be used to determine
+      // the delta since the last release.
+      .get(
+        '/repos/googleapis/java-trace/pulls?state=closed&per_page=100&sort=merged_at&direction=desc'
+      )
+      .reply(200, [
+        {
+          base: {
+            label: 'googleapis:master',
+          },
+          head: {
+            label: 'googleapis:release-v0.20.3',
+            sha: 'da6e52d956c1e35d19e75e0f2fdba439739ba364',
+          },
+          merged_at: new Date().toISOString(),
+        },
+      ])
+      .post('/graphql')
+      .reply(200, {
+        data: graphql,
+      })
+      // finding pom.xml files
+      .get('/search/code?q=filename%3Apom.xml+repo%3Agoogleapis%2Fjava-trace')
+      .reply(200, {
+        total_count: 1,
+        items: [{name: 'pom.xml', path: 'pom.xml'}],
+      })
+      // finding build.gradle files
+      .get(
+        '/search/code?q=filename%3Abuild.gradle+repo%3Agoogleapis%2Fjava-trace'
+      )
+      .reply(200, {
+        total_count: 0,
+        items: [],
+      })
+      // finding dependencies.properties files
+      .get(
+        '/search/code?q=filename%3Adependencies.properties+repo%3Agoogleapis%2Fjava-trace'
+      )
+      .reply(200, {
+        total_count: 0,
+        items: [],
+      })
+      // check for CHANGELOG
+      .get(
+        '/repos/googleapis/java-trace/contents/CHANGELOG.md?ref=refs%2Fheads%2Fmaster'
+      )
```

Comment:
@chingor13 if it's too much work fo this PR, we don't have to do it, but how do you feel about this refactor to how we do testing:

https://github.com/googleapis/release-please/blob/master/test/releasers/simple.ts

I think it cuts down on the giant `nock` stubs, which I like.

Reply:
Let's do it in a follow-up PR